core/rawdb/accessors_chain.go

```diff
@@ -682,11 +673,11 @@ type receiptLogs struct {
 
 // DecodeRLP implements rlp.Decoder.
 func (r *receiptLogs) DecodeRLP(s *rlp.Stream) error {
-	var stored storedReceiptRLP
-	if err := s.Decode(&stored); err != nil {
+	var rs types.ReceiptForStorage
+	if err := s.Decode(&rs); err != nil {
```

Comment:
Might be even better to call `rs.DecodeRLP` directly here, it saves an indirection.